vue/components/ui/molecules/search-persistent/save-filter-modal.vue

```diff
@@ -95,25 +121,148 @@ export const SaveFilterModal = {
             return (
                 this.searchData &&
                 this.filterNameData &&
-                Object.keys(this.tenacyValuesData).length > 0
+                Object.keys(this.tenancyCheckboxValuesData).length > 0 &&
+                this.isTenancyFormValid
             );
         },
         filters() {
-            return Object.keys(this.tenacyValuesData).map(tenancy => ({
+            return Object.keys(this.tenancyCheckboxValuesData).map(tenancy => ({
                 name: this.filterNameData,
                 value: this.searchData,
-                tenancy: tenancy,
+                tenancy: this.getSelectedTenancy(tenancy),
                 context: "" // TODO
             }));
+        },
+        isTenancyEmpty() {
+            return this.tenancyItemsData.length === 1;
+        },
+        isTenancyFormValid() {
+            for (const item of this.tenancyItemsData) {
+                if (
+                    this.isTenancyChoiceSelected(item.value) &&
+                    this.getSelectedTenancy(item.value) === null
+                ) {
+                    return false;
+                }
+            }
+
+            return true;
+        },
+        userTenancy() {
+            return { choices: [], selectedValue: "user", selectVisible: false };
+        },
+        builtTenancies() {
+            const builtTenancies = {};
+
+            this.tenancyItems.forEach(item => {
+                if (item.value === "user") builtTenancies[item.value] = this.userTenancy;
+                else {
+                    builtTenancies[item.value] = {
+                        choices: [],
+                        selectedValue: null,
+                        selectVisible: false
+                    };
+                }
+            });
+
+            return builtTenancies;
         }
     },
-    mounted: async function() {},
+    mounted: async function() {
+        this.tenancies = this.builtTenancies;
+
+        // Getting brands, channels and factories
+        this.tenancies.brand.choices = await this.getBrands();
+        this.tenancies.channel.choices = await this.getChannels();
+        this.tenancies.factory.choices = await this.getFactories();

```

Comment:
I understand you're pushing for code reuse with this approach, but I got the extreme feeling that the solution got over-engineered because of this. On the one hand you're trying to make this general for N tenancy types, but that premise is broken since you have to hard-code the `getX` anyway.

Split this into `brandChoices`, `channelChoices`, etc. Then every time you need to make common assertions, you just need to pass `brandChoices` instead of passing `"brand"` and then acessing `tenancies`.

Trust me that once you make this seemingly small change the overall structure will get much simpler.

Reply:
This was the code before I started simplifying things: https://github.com/ripe-tech/ripe-components-vue/commit/02789d3c04a30b93ab02c2bfcba1d75a1fdb1bf7 and https://github.com/ripe-tech/ripe-components-vue/commit/6dd56baa76942bfa183a7b5a0258ba8c74118b90

I changed it because I thought it was a little confusing to read the code and I felt like I had a lot of similar logic, so I tried to make simplify this and make it more reusable.

I made the changes so that I could just use a computed like the `builtTenancies`to create the data structures needed. After that I would just need to worry about how to load the choices and the code would work.